### PR TITLE
feat(api): treat sub and email fields of oidc tokens as distinct things; enable SA mappings by either

### LINKED ISSUE #1278: Consult `rbac.kargo.akuity.io/email` as a scope
# Proposed Feature

In v0.3.0, we will support mapping of `sub` and `groups` OIDC fields:

```yaml
metadata:
  annotations:
    rbac.kargo.akuity.io/sub: jessesuen
    rbac.kargo.akuity.io/groups: myorg:kargoadmin
```

But we should also support `email`, which is a standard OIDC scope.

# Motivation

`sub` is often an opaque user ID. But emails are not. In order to improve the UX of giving individuals access, we need to support the `email` scope.

# Suggested Implementation

Similar implementation as sub:

https://github.com/akuity/kargo/blob/4e28102659e94c151c751ad86344a9f40f829882/internal/kubeclient/indexer.go#L307-L330

### PR DESCRIPTION
**Pull request by krancour**
Fixes #1278

### CODE REVIEW

File: internal/api/kubernetes/client.go [hunk 1]
```diff
@@ -553,7 +553,7 @@ func getAuthorizedClient(globalServiceAccountNamespaces []string) func(
 			Name:        key.Name,
 		}
 
-		if userInfo.Username != "" {
+		if userInfo.Subject != "" {
```
Comment (gdsoumya) [inline internal/api/kubernetes/client.go:556]: Do we need to consider Email too?

File: internal/api/option/auth.go [hunk 2]
```diff
@@ -260,17 +259,19 @@ func (a *authInterceptor) WrapStreamingHandler(
...
 func (a *authInterceptor) listServiceAccounts(
 	ctx context.Context,
-	username string,
-	groups []string,
+	c claims,
 ) (map[string]map[types.NamespacedName]struct{}, error) {
 	queries := []libClient.MatchingFields{
 		{
-			kubeclient.ServiceAccountsBySubjectIndexField: username,
+			kubeclient.ServiceAccountsByOIDCEmailIndexField: c.Subject,
```
Comment (jessesuen) [inline internal/api/option/auth.go:266]: Is this a bug? Should this be c.Email?

File: internal/api/kubernetes/client.go (hunk 1 above)
Comment (krancour) [reply internal/api/kubernetes/client.go:556]: Good question. "sub" and "email" are both standard claims. We'll either have a value for both or for neither. So we only need to check one to know what kind of scenario we're dealing with.

File: internal/api/option/auth.go (hunk 2 above)
Comment (krancour) [reply internal/api/option/auth.go:266]: 😓 

Nice catch.

Comment (krancour) [reply internal/api/option/auth.go]: Fixed in b5ad67caa1c15c894428e63020fe1bcdee1093b5

**Review by jessesuen — APPROVED** [review]
LGTM!

**Pull request merged**